featuretools/entityset/serialization.py

```diff
@@ -1,175 +1,228 @@
-# -*- coding: utf-8 -*-
-
 import json
 import os
 import shutil
-import sys
-import uuid
-from tempfile import mkdtemp
 
-import numpy as np
 import pandas as pd
-from pandas.io.pickle import read_pickle as pd_read_pickle
-from pandas.io.pickle import to_pickle as pd_to_pickle
+
+from .. import variable_types
+
+FORMATS = ['csv', 'pickle', 'parquet']
+SCHEMA = {
+    'entities': ['id', 'index', 'time_index', 'variables', 'properties', 'loading_info'],
+    'relationships': {
+        'parent': ['parent_entity', 'parent_variable'],
+        'child': ['child_entity', 'child_variable']
+    }
+}
+VARIABLE_TYPES = {
+    getattr(variable_types, type).type_string: getattr(variable_types, type) for type in dir(variable_types)
+    if hasattr(getattr(variable_types, type), 'type_string')
+}
 
 
-def read_parquet(path, load_data=True):
-    '''Load an EntitySet from a path on disk, assuming
-    the EntitySet was saved in the parquet format.
+def to_entity_description(entity):
+    '''Serialize entity to data description.
+
+    Args:
+        entity (Entity) : Instance of :class:`.Entity`.
+
+    Returns:
+        dictionary (dict) : Description of :class:`.Entity`.
     '''
-    return read_entityset(path, load_data=load_data)
+    description = {}
+    for key in SCHEMA.get('entities'):
+        if key in ['id', 'index', 'time_index']:
+            description[key] = getattr(entity, key)
+        elif key == 'properties':
+            description[key] = {
+                'secondary_time_index': entity.secondary_time_index,
+                'last_time_index': entity.last_time_index is not None,
+            }
+        elif key == 'variables':
+            description[key] = [variable.create_data_description() for variable in entity.variables]
+        elif key == 'loading_info':
+            index = entity.df.columns.isin([variable.id for variable in entity.variables])
+            dtypes = entity.df[entity.df.columns[index]].dtypes.astype(str).to_dict()
+            description[key] = {'params': {}, 'properties': {'dtypes': dtypes}}
+        else:
+            raise ValueError('"{}" is not supported'.format(key))
+    return description
+
+
+def to_relationship_description(relationship):
```

Comment:
same here as `to_entity_description` we can get rid of the usage of schema

Reply:
Change applied in [this commit.](https://github.com/Featuretools/featuretools/pull/361/commits/a642a4db72be52203ab6c05fc9d580c462885389)